### PR TITLE
Updating OS Package metadata: license and description

### PR DESCRIPTION
**Pull request by doggydogworld**
Updates our OS packages (.rpm, .deb, etc) metadata to be up-to-date. Previous to this change it contained out-of-date information for our license and description.

changelog: Our OSS OS packages (rpm, deb, etc) now have up-to-date metadata.

### CODE REVIEW

**Review by fheinecke — APPROVED** [review]
LGTM provided that this builds/releases without issue

**Review by r0mant — APPROVED** [review]

File: build.assets/build-package.sh [hunk 1]
```diff
@@ -74,9 +74,9 @@ FPM_IMAGE_RPM="public.ecr.aws/gravitational/fpm:centos8-1.15.1-1"
 
 # extra package information for linux
 MAINTAINER="info@goteleport.com"
-LICENSE="Apache-2.0"
+LICENSE="AGPL-3.0-or-later"
```
Comment (zmb3) [inline build.assets/build-package.sh:77]: The packages we distribute are governed by `build.assets/LICENSE-community`, NOT by AGPLv3.

We should change this (is there a way to use a custom license in a package?), and we should also be including a copy of `LICENSE-community` in the package itself.

**Review by zmb3 — CHANGES_REQUESTED** [review]
See comment.

Also note that the licensing rules changed with v16, so this should _not_ be backported to v15.

File: build.assets/build-package.sh (hunk 1 above)
Comment (doggydogworld) [reply build.assets/build-package.sh:77]: Ah ok the task we had wasn't entirely clear on what was necessary. Thanks for the additional context.

We should be able to use a custom license. Afaik this just takes in an arbitrary string. Unfortunately there's not really a way to query for the contents of the license itself. Maybe we could provide a link to the license in the description?

Adding the `LICENSE-community` should be easy enough though. Just need to update `build-archive` to include it in the tarball.

Comment (zmb3) [reply build.assets/build-package.sh:77]: I suppose its even a little more complicated than that, as the `-ent` packages are not governed by `LICENSE-community`.

**Review by zmb3 — APPROVED** [review]

**Comment by doggydogworld** [comment]
Dev build: https://github.com/gravitational/teleport.e/actions/runs/12205154596#artifacts

# Tarball
```
tar -tvf ~/Downloads/teleport-v17.0.0-dev.gus-pkg-metadata.2-darwin-amd64-bin.tar.tar
drwxr-xr-x  0 runner staff       0 Dec  6 14:02 teleport/
-rwxr-xr-x  0 runner staff    2038 Dec  6 14:02 teleport/install
-rwxr-xr-x  0 runner staff 89892224 Dec  6 14:02 teleport/tbot
...
-rw-r--r--  0 runner staff     11896 Dec  6 14:02 teleport/LICENSE-community
```

# RPM
```
rpm -qip /work/teleport-17.0.0-dev.gus-pkg-metadata.2-1.arm.rpm
warning: /work/teleport-17.0.0-dev.gus-pkg-metadata.2-1.arm.rpm: Header V4 RSA/SHA512 Signature, key ID 2f67ad73: NOKEY
Name        : teleport
Version     : 17.0.0_dev.gus_pkg_metadata.2
Release     : 1
Architecture: armv7hl
Install Date: (not installed)
Group       : default
Size        : 589293987
License     : Teleport Community Edition License
Signature   : RSA/SHA512, Fri Dec  6 19:59:07 2024, Key ID 1b3d7a692f67ad73
Source RPM  : teleport-17.0.0_dev.gus_pkg_metadata.2-1.src.rpm
Build Date  : Fri Dec  6 19:57:58 2024
Build Host  : a4bf7ec79ff7
Relocations : /
Packager    : info@goteleport.com
Vendor      : Gravitational
URL         : https://goteleport.com/docs
Summary     : Teleport provides on-demand, least-privileged access to your infrastructure, on a foundation of cryptographic identity and zero trust, with built-in identity and policy governance [32-bit ARM Open source edition]\n\nDistributed under the Teleport Community Edition License
Description :
Teleport provides on-demand, least-privileged access to your infrastructure, on a foundation of cryptographic identity and zero trust, with built-in identity and policy governance [32-bit ARM Open source edition]\n\nDistributed under the Teleport Community Edition License
```

# deb

```
dpkg-deb -I teleport_17.0.0-dev.gus-pkg-metadata.2_arm.deb
 new Debian package, version 2.0.
 size 158822666 bytes: control archive=1379 bytes.
     577 bytes,    12 lines      control
     419 bytes,     7 lines      md5sums
    1213 bytes,    47 lines   *  postinst             #!/bin/sh
 Package: teleport
 Version: 17.0.0-dev.gus-pkg-metadata.2
 License: Teleport Community Edition License
 Vendor: Gravitational
 Architecture: armhf
 Maintainer: info@goteleport.com
 Installed-Size: 575482
 Provides: teleport
 Section: default
 Priority: optional
 Homepage: https://goteleport.com/docs
 Description: Teleport provides on-demand, least-privileged access to your infrastructure, on a foundation of cryptographic identity and zero trust, with built-in identity and policy governance [32-bit ARM Open source edition]\n\nDistributed under the Teleport Community Edition License
```

**Pull request merged**